### PR TITLE
Remove deprecated KotlinDslOptions.experimentalWarning

### PR DESCRIPTION
**Pull request by eskatos**
* https://github.com/gradle/gradle/issues/21907

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1280,7 +1280,12 @@
       </component>
       <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-plugins" version="3.0.0">
          <artifact name="gradle-kotlin-dsl-plugins-3.0.0.jar">
-            <sha256 value="4ba5a8716d81dde001ff3df916b11a523bb8a05c0a707de25a08fed9fc11c13f" origin="Generated by Gradle because artifact wasn't signed"/>
+            <sha256 value="4ba5a8716d81dde001ff3df916b11a523bb8a05c0a707de25a08fed9fc11c13f" origin="Artifact is not signed"/>
+         </artifact>
+      </component>
+      <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-plugins" version="3.1.0">
+         <artifact name="gradle-kotlin-dsl-plugins-3.1.0.jar">
+            <sha256 value="4667995032a32315ff331fc1e80226788b0f935a42c3399d7214e16d8c5a0419" origin="Artifact is not signed"/>
```
Comment (ljacomet) [inline gradle/verification-metadata.xml:1288]: We should really start signing those!

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -1280,7 +1280,12 @@
       </component>
       <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-plugins" version="3.0.0">
          <artifact name="gradle-kotlin-dsl-plugins-3.0.0.jar">
-            <sha256 value="4ba5a8716d81dde001ff3df916b11a523bb8a05c0a707de25a08fed9fc11c13f" origin="Generated by Gradle because artifact wasn't signed"/>
+            <sha256 value="4ba5a8716d81dde001ff3df916b11a523bb8a05c0a707de25a08fed9fc11c13f" origin="Artifact is not signed"/>
```
Comment (ljacomet) [inline gradle/verification-metadata.xml:1283]: Can't we remove that entry?

**Review by ljacomet — APPROVED** [review]
LGTM

Please double check removing the entry from the verification metadata

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (eskatos) [reply gradle/verification-metadata.xml:1283]: We can once the wrapper is updated

**Comment by eskatos** [comment]
@bot-gradle test and merge

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (eskatos) [reply gradle/verification-metadata.xml:1288]: Yes, will make sure it happens

**Pull request merged**